doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -118,10 +130,10 @@ Other tokens
   without using :cmd:`Notation` (omitting 25 specialized tokens that begin with
   ``#int63_``)::
 
-    ! #[ % & ' ( () (bfs) (dfs) ) * ** + , - ->
+    ! #[ % & ' ( () ) * + , - ->
```

Comment:
Intentional to remove `**`?

Reply:
As far as I could see, `**` is only used in `intros`, so I moved it to the tactics chapter.

Reply:
Why move it to the tactics chapter?  Even though this is at the moment in the Gallina chapter, it is the primary discussion of keywords and tokens, etc.  Tactics are always loaded (right?), so moving it splits up the info for no benefit.

I can live with putting things specific to optionally-loaded plugins into their chapters, but I would probably repeat some or all of that info here with some additional description.  Otherwise it's hard to find--you have to know/remember that the tokens and keywords depend on what's loaded and then you have to jump to multiple sections to figure out what list applies to your situation.

BTW, how often do people use the `-nois` option?

Reply:
I don't think it brings a useful information to mention `**` in the Gallina chapter. Yes, it is defined in the initial state, but in the initial state, you also have `||`, `&&`, `.1`, `.2`, `^`, `~`, `<->`, `++`, `::`, `/\`, `\/`, and if you require `Utf8`, you will have another bunch of keywords, etc., and the list will most probably grow any time you require another library.

I'm not saying that it should be moved to the tactics chapter either actually. Probably, a more scalable approach would simply to provide a command `Print Keywords` which tells what are the keywords defined in the current environment.

Also, I don't desperate that we will have one day a parser so that lexing is done at parsing time and the question of whether `||` is a proper keyword or is just twice `|` does not matter (I already have a propotype for that which needs to be pushed further).

Reply:
> BTW, how often do people use the `-nois` option?

Not so frequently today, but it is no reason to make the doc ignorant of this possibility.

Reply:
@jfehrle: How did you get the list of keywords? Was it by analyzing the `*.mlg`?

There is otherwise a way to list of keywords from `coqtop.byte`: do `Drop.`, then, under the OCaml toplevel, do `List.iter (fun s -> Printf.printf "%s " s) (CString.Set.elements (CLexer.keywords ()));;`. This will give the list of keywords in the current session.

Without `-nois`, it is:
```
! #[ #float64_abs #float64_add #float64_classify #float64_compare
 #float64_div #float64_eq #float64_frshiftexp #float64_ldshiftexp
 #float64_le #float64_lt #float64_mul #float64_next_down
 #float64_next_up #float64_normfr_mantissa #float64_of_int63
 #float64_opp #float64_sqrt #float64_sub #float64_type #int63_add
 #int63_addc #int63_addcarryc #int63_addmuldiv #int63_compare
 #int63_div #int63_div21 #int63_diveucl #int63_eq #int63_head0 
#int63_land #int63_le #int63_lor #int63_lsl #int63_lsr #int63_lt 
#int63_lxor #int63_mod #int63_mul #int63_mulc #int63_sub #int63_subc 
#int63_subcarryc #int63_tail0 #int63_type
 % & && ' ( () (bfs) (dfs) ) * ** + ++ , - ->
 . .( .. ... / /\ : :: ::= :< := :> :>> ;
 < <+ <- <-> <: <<: <= <> = => > >-> >= ? @ @{ 
Axiom CoFixpoint Definition Fixpoint Hypothesis IF Parameter Prop SProp Set 
Theorem Type Variable 
[ [= \/ ] ^ _ `( `{ 
as at by cofix discriminated else end exists exists2 fix for forall fun if 
in lazymatch let match multimatch return then using where with { {| | |- || } ~ -
```
With `-nois`, it is:
```
! #[ #float64_abs #float64_add #float64_classify #float64_compare 
#float64_div #float64_eq #float64_frshiftexp #float64_ldshiftexp 
#float64_le #float64_lt #float64_mul #float64_next_down 
#float64_next_up #float64_normfr_mantissa #float64_of_int63 
#float64_opp #float64_sqrt #float64_sub #float64_type #int63_add 
#int63_addc #int63_addcarryc #int63_addmuldiv #int63_compare 
#int63_div #int63_div21 #int63_diveucl #int63_eq #int63_head0 
#int63_land #int63_le #int63_lor #int63_lsl #int63_lsr #int63_lt 
#int63_lxor #int63_mod #int63_mul #int63_mulc #int63_sub #int63_subc 
#int63_subcarryc #int63_tail0 #int63_type
 % & ' ( () ) * + , - -> . .( .. / : :< := :> :>> ;
 < <+ <- <: <<: <= = => > >-> ? @ @{ 
Axiom CoFixpoint Definition Fixpoint Hypothesis Parameter Prop SProp Set 
Theorem Type Variable
[ ] _ `( `{ 
as at cofix discriminated else end fix for forall fun
if in let match return then where with { {| | } -
```

Note that none of the `#float` and `#int63` need to be keywords actually.  Do you know the point of adding a `#` to these names? Why not simply `float64_abs`, `float64_add` so that we can a `IDENT` in front of them to remove them from the keyword list?

Reply:
Yeah, I think I modified the code to print out the same data structure.  I haven't used Drop, I should do that sometime.

You're preaching to the choir :-).  It appears they're only used to set more user-friendly names in PrimFloat.v.  Seems to me changing this probably wouldn't be too disruptive.  And AFAIK the feature is relatively new (last year or so).

Reply:
I don't know what those `#` are for. It seems that we could remove them without issue, see #12122

Reply:
They're used to identify new-style attributes.  You can grep in the `.mlg` files:

```
  quoted_attributes:
    [ [ "#[" ; a = attribute_list ; "]" -> { a } ]
    ]
  ;
```

Reply:
Pierre was talking about the `#` in primitive tokens.

Reply:
OK.  If someone gets rid of the `#` then maybe they could also make all of the `#int63*` keywords into identifiers, if that's not too hard.

Reply:
That's precisely the goal of #12122 ;)